### PR TITLE
51 search result component

### PR DESCRIPTION
**Pull request by Bakeza**

### CODE REVIEW

File: client/src/components/Navbar/Navbar.tsx [hunk 1]
```diff
@@ -8,8 +12,27 @@ import {
...
+        pathname: '/search',
+        search: `${createSearchParams({ category: event, q: inputValue })}`,
+      })
+    } else {
+      setSearchParams({
+        category: event,
+        q: inputValue,
+      })
+    }
+  }
```
Comment (aaamra) [inline client/src/components/Navbar/Navbar.tsx:35]: just a suggestions
```suggestion
  const [category, setCategory] = useState<string>('event')
  const [search, setSearch] = useState<string>('')
  // eslint-disable-next-line no-unused-vars
  const [_, setSearchParams] = useSearchParams()

  const handleSearch = ():void => {
    if (location.pathname !== '/search') {
      navigate({
        pathname: '/search',
        search: `${createSearchParams({ category, q: search })}`,
      })
    } else {
      setSearchParams({
        category,
        q: search,
      })
    }
  }
```

File: client/src/components/SearchResult/index.tsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+                  backgroundColor: '#256D85',
+                  opacity: [0.9, 0.8, 0.7],
+                },
+              }}
+              size="small"
+            >
+              Follow
+            </Button>
+          </CardActions>
+        </Card>
```
Comment (aaamra) [inline client/src/components/SearchResult/index.tsx:93]: i think you can extract this to a separate component

File: client/src/components/SearchResult/index.tsx [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+                  backgroundColor: '#256D85',
+                  opacity: [0.9, 0.8, 0.7],
+                },
+              }}
+              size="small"
+            >
+              Join
+            </Button>
+          </CardActions>
+        </Card>
```
Comment (aaamra) [inline client/src/components/SearchResult/index.tsx:162]: and this one

File: client/src/components/SearchResult/index.tsx [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
```
Comment (aaamra) [inline client/src/components/SearchResult/index.tsx:233]: and this one

File: client/src/services/ApiService.ts [hunk 5]
```diff
@@ -10,6 +10,7 @@ class ApiService {
 
   public static init(): void {
     this.axios.defaults.baseURL = process.env.REACT_APP_BASE_URL
+    // this.axios.defaults.baseURL = 'http://localhost:8080'
```
Comment (aaamra) [inline client/src/services/ApiService.ts:13]: remove this comment

**Review by aaamra — CHANGES_REQUESTED** [review]

File: client/src/components/SearchResult/EventCardHandler.tsx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+        { ` | Time: ${dayjs(startTime).format('HH:mm:ss a')}`}
+      </Typography>
+      <Typography variant="body2" color="text.secondary">
+        {description}
+      </Typography>
+      <Typography
+        className={status === 'in-progress'
+          ? 'green' : status === 'upcoming'
+            ? 'orange' : status === 'closed'
+              ? 'red' : 'transparent'}
```
Comment (aaamra) [inline client/src/components/SearchResult/EventCardHandler.tsx:69]: i think you can extract this logic to a function to make it more readable

something like that
```js
const getStatusClassName = (status) => {
  if(status === 'in-progress'){
     return 'green'
  } else if () {
    /// etc
  } else {
     //etc
  }
}
```

File: client/src/components/SearchResult/FriendCardHandler.tsx [hunk 7]
```diff
@@ -0,0 +1,75 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/require-default-props */
```
Comment (aaamra) [inline client/src/components/SearchResult/FriendCardHandler.tsx:2]: don't disable eslint

File: client/src/components/SearchResult/FriendCardHandler.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  Typography,
+  Button,
+} from '@mui/material'
+
+interface cardData{
+  image?: string
+  username?: string
+  bio?: string
+  button?: string
+}
```
Comment (aaamra) [inline client/src/components/SearchResult/FriendCardHandler.tsx:19]: the same interface issue here

File: client/src/components/SearchResult/FriendCardHandler.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+const FriendCardHandler:FC<cardData> = ({
```
Comment (aaamra) [inline client/src/components/SearchResult/FriendCardHandler.tsx:21]: also the same for component name

File: client/src/interfaces/ISearchResult.tsx [hunk 10]
```diff
@@ -0,0 +1,6 @@
+export default interface ISearchResult{
+  category?: string
+  input?: string
+  setCategory?:(c:string)=> void
+  setInput?:(c:string)=> void
+}
```
Comment (aaamra) [inline client/src/interfaces/ISearchResult.tsx:6]: I think we don't use this interface
so you can delete it

**Review by aaamra — CHANGES_REQUESTED** [review]

**Review by aaamra — APPROVED** [review]

**Review by SaifHayek — APPROVED** [review]

**Review by MostafaBalousha123 — APPROVED** [review]

**Pull request merged**